### DISCUSSION TITLE
Welcome to Computational Mechanics Summer '23

### DISCUSSION THREAD

**Discussion by cooperrc** [cooperrc/computational-mechanics#187] [General]
Greetings, 
Welcome to our Summer session of Computational Mechanics! 

We are starting a fast-paced computational mechanics course. Let's get to know each other today. 
- _Who_ are you? and _Why_ are you here in ME3255?
- What did you do for Memorial Day weekend?
- Show us an example code in Python and tell us what it does. 
    _use the ` ```python - ``` ` markdown format to show code e.g._

````
```python 
x = 1
print(x)
```
````

becomes

```python
x = 1
print(x)
```

**Comment by cooperrc** [comment]
### who are you 
Prof. Cooper, I've been a Mechanical Engineering professor here at UConn for 7 years. Its been an amazing journey and I've got lots of plans for my future here. One of them is to build better and more accessible computational engineering tools for us. 

### What did you do for Memorial Day weekend
I went back to East Lyme, CT for my 20th high school reunion. Which seemed crazy that so much time had passed. 

### Show an example code

```python
import numpy as np
t = np.linspace(0, 1, 50)
y = 5 - 0.5*9.81*t**2
print(np.mean(y))
```

This code creates 50 time steps between 0 sec and 1 sec as `t`. Then, I predict the height of a falling object released from rest at 5 meters, `y` = $5 - \frac{g}{2}t^2$. Finally, I calculate the average height of the object over the 1-second time frame. 

_Physics fun Q: what do you think the avg height is? Max and min are 5 m and 0.095 m_

**Reply by predwood** [reply]
$$ \text{maybe this, avg height} = \frac{1}{1 - 0} \int_{0}^{1} (5 - 0.5 \cdot 9.81 \cdot t^2) \ dt $$

**Comment by GregCarr658** [comment]
**Who are you?**
My name is Greg Carr I am am a rising senior at UConn studying mechanical engineering. I'm not sure what exactly I want to do with my career but I am fascinated by the design and manufacturing process related to the degree. I am here in ME 3255 to get a head start on the require courses I have left to take my senior year.

**What did you do for Memorial Day weekend?**
For my memorial day I stayed at home and relaxed. We did do some grilling and enjoyed the nice weather outside.

**Example Python code**
It has been a while since I used python but the code bellow's purpose is to return a random color from the list of 4 colors provided.

 ```python -
import random

def get_random_color():
      colors = [ 'red', 'blue' , 'yellow' , 'purple' ]
      random_color = random.choice(colors)
      return random_color
```

**Comment by n123wong** [comment]
**Who are you?** 
My name is Nicole Wong and I'm a rising senior at UConn, majoring in mechanical engineering and minoring in entertainment engineering. I'm taking ME3255 to fulfill a requirement for my major and to ease my courseload for senior year.

**What did you do for Memorial Day weekend?**
I spent the long weekend in New York City with my family. We spent most of the weekend sightseeing and got to see two shows on Broadway.

**Example Python Code**
The following code utilizes a for loop to iterate through a string, outputting one printed letter per iteration. 

``` python
for x in 'Python':
     print(x)
```

**Reply by cooperrc** [reply]
> I spent the long weekend in New York City with my family. We spent most of the weekend sightseeing and got to see two shows on Broadway.

Cool! where did you stay?

**Comment by naysajg** [comment]
**Who are you?**
My name is Naysa Joseph-Gabriel and I'm a rising senior double majoring in Biomedical Engineering and French. I'm taking Computational Mechanics this summer as my track elective to fulfill that requirement and to lighten up my courseload next semester.

**What did I do for Memorial Day weekend**
I had a very relaxing memorial day at home. My family invited some family friends we hadn't seen in years over for lunch.

**Python Code**
```python
import math
a= 2
b= 3
y= a*b
z= y**2
print(z)
```
This code outputs the product of a and b and squares it. In this case, the output, z, would be 36

**Comment by ahemchand** [comment]
**Who are you?**
Hello everyone, my name is Alex Hemchand and I am a rising junior majoring in mechanical engineering. I am taking computational mechanics this summer in order to fulfill a major requirement and to create more space in my schedule for upcoming semesters. I look forward to learning more engineering applications utilizing Python.

**What did you do for Memorial Day weekend?**
For Memorial Day weekend I went camping at Ketteltown State Park in Southbury with a group of friends. 

**Python Code**
```python
import numpy as np
a=3
b=4
c=(a**2+b**2)**0.5
print(c)
```
Above is a representation of Pythagorean theorem, utilized to calculate the hypothesis of a right triangle.

**Reply by cooperrc** [reply]
```python
import numpy as np
a=3
b=4
c=(a**2+b**2)**0.5
print(c)
```

one suggested update:

```python
import numpy as np
v = np.array([3, 4])
c=np.sum(v**2)**0.5
print(c)
```

It does the same thing, but now you can define `v` as any size vector (1-D, 2-D, 3-D, **N-** D!)

**Comment by mcapps97** [comment]
Who are you?

Hi, My name is Michael Capps, and I am taking ME 3255 as a requirement for my major (Mech Eng). I am doing it over the summer to lighten my load senior yet.

Over the weekend, I just did some barbequing. I made steak with grilled with corn and green beans (tossed in a lemon and garlic sauce), and then made a baked sausage and rigatoni dish the following day. 
``` print('I love Summer courses')```

This code will show how I feel about summer courses.

**Reply by mcapps97** [reply]
if this doesnt count then:

```import matplotlib.pyplot as plt
import math
import numpy as np



print()
print(' A:')
data_a = [-1.738, 0.145, 2.024, 3.868, 4.829, 6.690, 8.333]
n_a = len(data_a)
print('Size {}.'.format(n_a))
std_a = np.std(data_a, ddof = 1)
print('STD A {:.2f}.'.format(std_a)) ```

This takes a data set and finds the length prints it, then takes the standard deviation.

**Comment by RMaguire23** [comment]
**Who are you?**

Hi, my name is Ryan Maguire and I am a rising Senior studying Mechanical Engineering. I am taking ME3255 in order to catch up/graduate on time after taking a semester off. 

**What did I do for Memorial Day weekend**

For Memorial Day weekend I got to hangout with my friends who go to school in Florida, along with my other friends from home,  so it was good to see everyone and just have a good time.  

**Python Code**

```python 
x = 4
y = 7
z = 2
r = ((x*y)/z)
print (r)

 ```
This code takes in 3 variables (x, y, z), and then defines r which is the product of x and y over z, and then prints the solution.

**Comment by ewk19001** [comment]
**Who are you? and Why are you here in ME3255?**
Hi, my name is Evan Kluge and and I am a rising senior studying mechanical engineering. I am taking this class over the summer as part of my requirements and to free up space for the fall semester as I am taking a graduate level course to begin pursuing my Master's. 

**What did you do for Memorial Day weekend?**
I ended up going to New York for the weekend to hang out with some friends. I also went to the Yankees game where they hit a walk-off in extra innings.

**Show us an example code in Python and tell us what it does.
use the ```python - ``` markdown format to show code e.g.**
```python 
number1 = 5
number2 = 13
number3 = 4

if (number1 >= number2) and (number1 >= number3):
   biggest = number1
elif (number2 >= number1) and (number2 >= number3):
   biggest = number2
else:
   biggest = number3

print('The biggest number is', biggest)
```
This code takes in three numbers and outputs the greatest one.

**Comment by AndyMiele** [comment]
**Who are you?**
Hi, my name is Andy Miele, I'm a rising senior at Uconn Storrs, studying Mechanical engineering with a concentration in aerospace engineering. I am taking this course as part of my major requirements, in addition to lightening my course load for next semester.

**What did you do for Memorial Day weekend?**
I stayed home for Memorial Day weekend, went to a friends brothers graduation party, had a barbeque at home with my family, and hung out with friends. I'm also in the process of building an electric guitar, so I spent a lot of time working on that as well.

**Show us an example code in Python**
```python 
def average(a,b):
    c = a + b
    avg = c/2
    print("The average of", a, "and", b ,"is:" avg)
```
This code takes two integers as input, then finds the average of them.

**Comment by gunnardesantisUCONN** [comment]
**Who are you? and Why are you here in ME3255?**

Gunnar DeSantis, I'm a 3rd year at UVA but I'm taking this class so I can graduate on time. I'm MechE with an Applied Math minor.  I live in Redding, CT and am interning at ASML this summer.

**What did you do for Memorial Day weekend?**
Wakeboarding, a couple barbeques, a 5k, work


**Show us an example code in Python and tell us what it does.**
```python 
E = 200000000000.0 #Young's Modulus (Pa)
d1 = 0.01 #ID (meters)
d2 = 0.04 #OD (meters)
l = 1.0 #length (meters)

print("The cylinder has an inner diameter of ", d1, ", an outer diameter of ", d2, ", and a length of ", l, ".")
A = (3/4) * (d2*d2-d1*d1)
k = E*A/l
print("The stiffness is ", k, ".") 
```
This code gives the stiffness of a solid cylinder of a material!

**Reply by cooperrc** [reply]
> I'm a 3rd year at UVA

Welcome back to the Northeast! 

I like the stiffness calculator, but I noticed you're using $\pi \approx 3$. Nothing wrong with the approximation, but might want to let unsuspecting young engineers know ;)

**Comment by christian-villandry** [comment]
**Who are you? and Why are you here in ME3255**?
My name is Christian Villandry. I am a rising senior majoring in mechanical engineering. I am taking this course as a major requirement to help myself get ahead to focus more on senior design over the following semesters. I really look forward to learning more about python.
What did you do for Memorial Day weekend?
I went on a long motorcycle ride with my dad to visit family in Rhode Island.
Show us an example code in Python and tell us what it does.
```python -
def is_even(n):
    if n % 2 == 0:
         return True
    else:
         return False
```
This function returns a 'True' if the input is even and a 'False' if the input is odd.

**Comment by kguert** [comment]
**Who are you?** 
My name is Kyle Guertin, I'm a rising senior mechanical engineering student. I play drums in a few bands, and I enjoy working out, cooking, and spending time with my family, wife, and dogs.

**Why are you here in ME3255?**
I am here for the ME requirement for my degree, but I also needed to take this class in order to enroll in classes for the fall semester if I plan to graduate in the Spring. 

**What did you do for Memorial Day weekend?**
I hung out and grilled some food with my parents, brothers, nephew, and niece, played some music with friends, and took the dogs on a couple of adventures.

**Show us an example code in Python and tell us what it does.**

```python 
n1 = int(input("Enter Integer 1: "))
n2 = int(input("Enter Integer 2: "))
n3 = int(input("Enter Integer 3: "))

maxInt = max(n1,n2,n3)
print("Largest Integer:",maxInt)
```

This Python Code takes three user inputs for integers and returns the integer with the maximum value.

**Comment by tad12002** [comment]
### Who are you? Why are you here in ME3255?
My name is Tyler Dickey, I'm a rising senior in mechanical engineering at UConn Storrs. My interests include mountain biking, riding dirtbikes and tinkering on cars. I enjoy working out and playing soccer as well. 

I'm here for a class requirement for my major. I decided to take this over the summer in order to lighten my course load this coming fall and spring semester.

### What did you do for Memorial Day weekend?
I went to go ride dirt bikes in Massachusetts with my dad and had a cookout with my family. 

### Show an example code
```python 
x = 4
y = 462

temp = x
x = y
y = temp

print('The value of x after swapping: {}'.format(x))
print('The value of y after swapping: {}'.format(y))
```

In this code the  temp variable holds the value of x so that the variable y can be swapped in with x then the temp back to y. Changing the associated value of x and y.

**Reply by cooperrc** [reply]
> My interests include mountain biking, riding dirtbikes and tinkering on cars

I love mountain biking (Mansfield is a great place for it). I haven't ridden a dirt bike in ~20 years though. It was fun

**Comment by jhille** [comment]
**Who are you**
I'm Josh, a senior mechanical engineering major. Some of my interests are cooking, growing plants, and frogs. I took this class to lighten up my course load for next semester.

**What did you do memorial day weekend**
I went to a music festival with a friend and spent memorial day having a bbq with family.

**Show an example code**
```
accuracy = input("larger number equals greater accuracy: ")

a = 2
b = 3
c = 4

nilakantha = 0
for i in accuracy:
    nilakantha = nilakantha + 4/(a*b*c)
    a = a + 2
    b = b + 2
    c = c + 2
    nilakantha = nilakantha - 4/(a*b*c)
    a = a + 2
    b = b + 2
    c = c + 2

print(3 + nilakantha)
```

This code uses the nilakantha series (π = 3 + 4 / (2*3*4) – 4 / (4*5*6) + 4 / (6*7*8) – . . .) to estimate the value of pi to the inputed degree of accuracy which controls how many calculations to make in the estimation

**Reply by cooperrc** [reply]
> nilakantha series approximation

Cool! I had to google that one, [Nilakantha series](https://www.geeksforgeeks.org/calculate-pi-using-nilkanthas-series/)

**Comment by samathrocket** [comment]
Who are you? 
My name is Samuel Athapaththu, and I am a senior at GMU. I am taking this course as a prerequisite for my senior capstone next semester. My plans for the future include continuing my work for the DOD as an engineer and possibly switching over and working on aviation at some point.  For break, I headed up to CT to visit some family and spent time unplugged with them. 

I haven't coded in python in a long time but this is a pretty simple code just to wave hi to yall!

Example Code

```python 
string1 = "Wave"
string2 = " Hi"
joined_string = string1 + string2
print(joined_string)
```

**Comment by JonCarlosRestrepo** [comment]
Hello, my name is Jon Carlos Restrepo Jr. I am currently attending this summer course to overall make my last year at UConn as a future senior smooth and lighter.

I am currently in study abroad and memorial day is not celebrated in Prague so for that day I just went to class and kayaked afterward.

The code below is to show that I like doughnuts
```python
a = 'I'                                              #a, b, and c are my variables
b = 'like'
c = 'Doughnuts'
Name = a + " " + b + " " + c         #Just combining variables and adding spaces to make a sentence
print(Name)
```

**Comment by ryanzwick** [comment]
Hello my name is Ryan Zwick. I run the cycling team here at Uconn and I love to ride my bike. I also like cars, wether its working on them or racing them. I'm here for a requirement of my major but also am super eager learn python.

I moved into a new apartment over the long weekend a little closer to campus and also got a couple of mountain bike rides in.

This code is to check your mountain bike tire pressure. 

```python - 
val = input("Enter your Mountain Bike tire pressure:")
val2 = int(val)

b = 20

if b < val2:
    print ("Your tire pressure is OK")
else: 
    print('Add more air.')
```

**Reply by cooperrc** [reply]
> This code is to check your mountain bike tire pressure.

Maybe we should reference rider weight, gravel-vs-rock, and tire size too ;)

[Silca pressure guide](https://silca.cc/pages/sppc-form)

might need to ask for the developer for the Python dev tools though

**Reply by ryanzwick** [reply]
Very cool! I haven't seen that one before but looks super useful for cyclocross and I might have to use that! Norco also has a program called [ride aligned](https://www.norco.com/bike-setup-guide/bikes/) that displays the perfect setup for everything based on your height and weight, it gives suspension pressures, tire pressures and even the ideal suspension rebound settings! I wish more company's had that, it really makes getting a base setup on a mountain bike so much easier especially with coil rear springs.

**Comment by smsandor** [comment]
Hi all, my name is Sarah. I am a Senior studying Mechanical Engineering. I am taking this course to graduate in August :)

For memorial day I had my friends over to my house in Narragansett for the long weekend. We went to the beach and had a bonfire.

```python
x = 'UConn'
y = 'School of'
z = 'Engineering'
College = x + " " + y + " " + z
print(College)
```

**Comment by MasudurR2002** [comment]
Hello, my name is Masudur Rahman. I am a rising senior in Mechanical Engineering at UCONN. I am excited to take ME 3255 as it is a requirement for my major.  

For Memorial Day weekend I visited my family in New York. I haven't seen them in over a year, so it was nice catching up. 

An example code: 

```python
import math
a= 3
b= 5
c= a+b
x= c**2
print(x)
```

This code outputs the sum of a and b, then it squares it. Therefore the output, x, would be 64.

**Reply by cooperrc** [reply]
the `math` library is pretty limited and in this case you don't really need it. In general, I use the `numpy` versions of `sum`, `mean`, `sin`, `cos`, etc.

**Comment by pannetta01** [comment]
who are you

My name is Patrick and I am a rising senior here at UConn. I am looking to take a summer class in order to make my senior year a little lighter. 

I had a family gathering on memorial day and went tubing with some cousins. 


Example code:

```python 
# Function to calculate the sum of two numbers
def calculate_sum(a, b):
    return a + b

# Main program
num1 = 5
num2 = 10
result = calculate_sum(num1, num2)
print("The sum of", num1, "and", num2, "is:", result)
```
This code defines a function called calculate_sum that takes two parameters a and b and returns their sum. In the main program, two numbers (num1 and num2) are initialized to 5 and 10, respectively.

**Comment by Nathangarala** [comment]
Who are you:

Hi I am Nathan and I'm going into my senior year in the Mechanical Engineering program here at UConn. I am taking this summer course to get ahead and make my senior year a little easier. 

For memorial day I took a trip down to Misquamicut Beach with some friends and then went out on a friends boat on Highland lake in Winsted. 

Here's a sample code:
```python
Age = 20
Weight = 180
Heart_Rate = 140
Time = 8
Calories = ((Age * 0.2757) + (Weight * 0.03295) + (Heart_Rate * 1.0781) - 75.4991) * Time / 8.368

print(f'Calories: {Calories:.2f} calories')
```
This code takes your age, weight, heart rate and time spent exercising and inputs them into a function to calculate the amount of calories burned during the exercise. This is then printed for the user to read.

**Reply by cooperrc** [reply]
Misquamicut is beautiful, but it was cold with that wind blowing. 

If you want a function instead of a script, you can use:

```python
def calorie_burn(Age = 20,
Weight = 180,
Heart_Rate = 140,
Time = 8):
  Calories = ((Age * 0.2757) + (Weight * 0.03295) + (Heart_Rate * 1.0781) - 75.4991) * Time / 8.368
  return Calories

print(f'Calories: {calorie_burn():.2f} calories')
```

**Comment by Engnearguy** [comment]
Who are you?
My name is Paolo Rassa and I am a rising Senior at UConn studying Mechanical Engineering. I am taking ME 3255 to fulfill a requirement for my major and to allow for some breathing room for my senior year.

What did you do this Memorial day Weekend?
For Memorial Day Weekend, I went out to a nearby beach with my family, where we relaxed and grilled all afternoon.

Give an example of Python Code and what it does:
```python
posdiv = list(range(1,100))

num = int(input("Give a number and I'll tell you if it is a prime number.\n\n"))

    
divisors = [x for x in posdiv if num % x == 0]
if len(divisors) == 2:
    print("It is a prime number!!!\n")
else:
    print("No, it is not a prime number.\n")
```
This code determines whether the number input is a prime number.

**Reply by cooperrc** [reply]
Nice work on the prime number checker. You might be limited to integers < 99*2 though because `posdiv` goes 1 -> 99.

**Comment by alexandria-b** [comment]
Who are you?

Hi, I am Alexandria, a rising mechanical engineering senior. I am taking this class to allow some space in my schedule the upcoming semesters.

Over the Memorial Day weekend, I hung out with friends and worked on a few knitting projects.

```python
x = 3
y = 4
z = 5

if (x < y) and (y < z):
print(‘y is between x and z, the letters are in order’)
else:
print('The letters are Not in order')
```
The code determines if the value of 'y' is greater than or less than 'x' and 'z' and the prints out whether the numbers are in order or not.

**Reply by cooperrc** [reply]
Cool, with Python it uses tabs or spaces for `if-else` statements. So you'll need to indent the `print` lines.

**Comment by baiquanchen** [comment]
Who are you? and Why are you here in ME3255?
Hi I am baiquan a ME student, I take this course it try to finish my required courses
What did you do for Memorial Day weekend?
ummm, nothing

def sum_numbers(input_list):
    return sum(input_list)

numbers = [1, 2, 3, 4, 5]
print(sum_numbers(numbers))

the code sum up all the numbers

**Comment by predwood** [comment]
**who are you**

Peter R., Mech. Eng. major at UConn Storrs, so my advisor suggested I take this course

**what did you do for Memorial Day weekend?**

I can't remember

**Example code**

```python

def drawCat():
    print (" /\_/\\\n( o.o ) \n > ^ <")
    

drawCat()

```
my code defines a callable function that prints a cat out of characters and newlines
